File: Solutions/0387/0387.js
var firstUniqChar = function(s) {
    const count = new Array(256).fill(0); // Expand array size for ASCII characters

  for (const c of s) {
    count[c.charCodeAt(0)]++; // Use charCodeAt for correct indexing
  }

  for (let i = 0; i < s.length; i++) {
    const charIndex = s.charCodeAt(i);
    if (count[charIndex] === 1) {
      return i;
    }
  }

  return -1; // No unique characters found
};

File: Solutions/0387/0387.java
class Solution {
    public int firstUniqChar(String s) {
        int[] count = new int[256]; // Use size 256 for ASCII characters

        for (char c : s.toCharArray()) {
            count[c]++;
        }

        for (int i = 0; i < s.length(); i++) {
            if (count[s.charAt(i)] == 1) {
                return i;
            }
        }

        return -1;
    }
}
